From: Wayne Boyer <wayneb@linux.vnet.ibm.com>
Date: Wed, 9 Jun 2010 08:24:55 -0700
Subject: [PATCH] [SCSI] ipr: add writeq definition if needed
X-Git: c5f10187965f93ef7ef67da9c7c449b13b6dee1b
References: FATE#311692
Patch-Mainline: 2.6.34

Compiling the driver will fail on 32 bit powerpc and other
architectures where writeq is not defined.  This patch adds a
definition for writeq.

Signed-off-by: Wayne Boyer <wayneb@linux.vnet.ibm.com>
Acked-by: Brian King <brking@linux.vnet.ibm.com>
Signed-off-by: James Bottomley <James.Bottomley@suse.de>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/ipr.h |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/drivers/scsi/ipr.h b/drivers/scsi/ipr.h
index ea391ff..0ef9a67 100644
--- a/drivers/scsi/ipr.h
+++ b/drivers/scsi/ipr.h
@@ -1860,4 +1860,12 @@ static inline int ipr_sdt_is_fmt2(u32 sdt_word)
 	return 0;
 }
 
+#ifndef writeq
+static inline void writeq(u64 val, void __iomem *addr)
+{
+        writel(((u32) (val >> 32)), addr);
+        writel(((u32) (val)), (addr + 4));
+}
 #endif
+
+#endif /* _IPR_H */
-- 
1.6.0.2

